azurerm/internal/services/network/parse/virtual_hub.go

```diff
@@ -1,23 +1,33 @@
-package network
+package parse
 
 import (
 	"fmt"
 
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 )
 
-type VirtualHubResourceID struct {
+type VirtualHubId struct {
 	ResourceGroup string
 	Name          string
 }
 
-func ParseVirtualHubID(input string) (*VirtualHubResourceID, error) {
+func NewVirtualHubID(resourceGroup, name string) VirtualHubId {
+	return VirtualHubId{
+		ResourceGroup: resourceGroup,
+		Name:          name,
+	}
+}
+func (id VirtualHubId) ID(subscription string) string {
```

Comment:
_minor:_ the other formatters use `subscriptionId` as the argument so we may as well make this consistent:

```suggestion
func (id VirtualHubId) ID(subscriptionId string) string {
```